### PR TITLE
Adds user-definable log formatting

### PR DESCRIPTION
**Pull request by SonicZentropy**
# Objective

- Allow users to alter log formatting

## Solution

- tracing_subscriber offers a Subscriber trait to allow this customization, so I put one into LogSettings and check for it before making the prior default Subscriber.

### CODE REVIEW

**Comment by SonicZentropy** [comment]
Post-CI I'm not sure this is even worth draft status at the moment.  It seems wasm and android builds both add required trait bounds on anything holding a Subscriber that enter the HRTB territory, which is fascinating but pretty beyond me at this point and perhaps better left to collect cobwebs until I understand what's going on better

**Comment by SonicZentropy** [comment]
Nevermind, found a better way, seems to work fine now!

**Comment by rparrett** [comment]
I wonder how this potentially interacts with the `trace_chrome` and `trace_tracy` features, and if it makes sense to use a separate subscriber for logging and tracing.

Could be a bit of a footgun if defining a custom logging format prevents you from profiling your code in the future.

I just want to point out another PR that touches this code here: <https://github.com/bevyengine/bevy/pull/3583>.

**Comment by SonicZentropy** [comment]
You actually were completely correct about that @rparrett, though I didn't see this until I just stumbled across it in use.  I stuck my head into tokio's Discord and had them help me modify it to a more proper version.  This one should take care of those issues

**Review by IceSentry — APPROVED** [review]

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_log/src/lib.rs [hunk 1]
```diff
@@ -46,12 +46,14 @@ use tracing_subscriber::{prelude::*, registry::Registry, EnvFilter};
 /// # use bevy_internal::DefaultPlugins;
 /// # use bevy_app::App;
 /// # use bevy_log::LogSettings;
-/// # use bevy_utils::tracing::Level;
+/// # use bevy_utils::tracing::{ Level };
```
Comment (mockersf) [inline crates/bevy_log/src/lib.rs:49]: this looks like something that was part of an experiment and you didn't revert back completely?

Comment (SonicZentropy) [reply crates/bevy_log/src/lib.rs:49]: Yes, sorry about the delay, honestly forgot this PR existed after a while.  That was probably my global config clashing and me not noticing, sorry

**Comment by SonicZentropy** [comment]
I missed the super basic formatting problem and this didn't get merged, apologies.  Looks like it has a conflict now, so I'll try to resolve it this weekend since I'm sure it'll be minor

File: crates/bevy_log/src/lib.rs (hunk 1 above)
Comment (alice-i-cecile) [inline crates/bevy_log/src/lib.rs:49]:
```suggestion
/// # use bevy_utils::tracing::Level;
```

**Comment by alice-i-cecile** [comment]
@SonicZentropy I'd like to merge this :) While you're rebasing, there's a tiny formatting suggestion for the doc example too.

**Comment by mvlabat** [comment]
It would be great if we allowed our users to pass not just 1 layer but multiple ones. Or we maybe could allow passing a custom subscriber, like it was done in the initial attempt, I'm curious what was the problem exactly with that one.
This way users can also get the benefit of not only customizing the formatter but also passing additional layers - that would come in handy for the [puffin integration](https://github.com/bevyengine/bevy/pull/4730) for instance if we decide that it should exist outside of Bevy and be added by users via either a plugin or by passing a puffin layer manually.

**Comment by tygyh** [comment]
I believe the functionality from this PR has already been added in other later changes. Is there a reason to keep it open?

**Pull request closed without merging**